tests/scripts/thread-cert/Cert_9_2_10_PendingPartition.py

```diff
@@ -176,6 +171,77 @@ def test(self):
+            lambda p: {TLV_REQUEST_TLV, NETWORK_DATA_TLV, ACTIVE_TIMESTAMP_TLV} <= set(p.mle.tlv.type))
+
+        # Step 7: Router MUST multicast a MLE Data Response
+        _rpkts.filter_ipv6_dst(LINK_LOCAL_ALL_NODES_MULTICAST_ADDRESS).filter_mle_cmd(
+            MLE_DATA_RESPONSE).must_next().must_verify(lambda p: {
+                SOURCE_ADDRESS_TLV, LEADER_DATA_TLV, NETWORK_DATA_TLV, ACTIVE_TIMESTAMP_TLV, PENDING_TIMESTAMP_TLV
+            } <= set(p.mle.tlv.type) and {NM_COMMISSIONER_SESSION_ID_TLV, NM_BORDER_AGENT_LOCATOR_TLV} <= set(
+                p.thread_meshcop.tlv.type) and p.thread_nwd.tlv.stable == [0])
+
+        # Step 8: MED MUST send a unicast MLE Data Request to Router_1,
+        with pkts.save_index():
+            pkts.filter_wpan_src64(MED).filter_wpan_dst64(ROUTER).filter_mle_cmd(MLE_DATA_REQUEST).must_next(
+            ).must_verify(lambda p: {TLV_REQUEST_TLV, NETWORK_DATA_TLV, ACTIVE_TIMESTAMP_TLV} <= set(p.mle.tlv.type))
+
+        # Step 9: Router MUST send a unicast MLE Data Response to MED_1
+        _rpkts.copy().filter_wpan_dst64(MED).filter_mle_cmd(MLE_DATA_RESPONSE).must_next().must_verify(lambda p: {
+            SOURCE_ADDRESS_TLV, LEADER_DATA_TLV, NETWORK_DATA_TLV, ACTIVE_TIMESTAMP_TLV, PENDING_OPERATION_DATASET_TLV
+        } <= set(p.mle.tlv.type) and {
+            NM_CHANNEL_TLV, NM_NETWORK_MESH_LOCAL_PREFIX_TLV, NM_PAN_ID_TLV, NM_DELAY_TIMER_TLV,
+            NM_ACTIVE_TIMESTAMP_TLV
+        } <= set(p.thread_meshcop.tlv.type) and p.thread_nwd.tlv.stable == [0])
+
+        # Step 10: Router MUST send MLE Child Update Request to SED_1
+        _rpkts.range(pkts.index).filter_wpan_dst64(SED).filter_mle_cmd(
+            MLE_CHILD_UPDATE_REQUEST).must_next().must_verify(lambda p: {
+                SOURCE_ADDRESS_TLV, LEADER_DATA_TLV, NETWORK_DATA_TLV, ACTIVE_TIMESTAMP_TLV, PENDING_TIMESTAMP_TLV
+            } <= set(p.mle.tlv.type))
+
+        # Step 11: SED MUST send a unicast MLE Data Request to Router_1
+        pkts.filter_wpan_src64(SED).filter_wpan_dst64(ROUTER).filter_mle_cmd(MLE_DATA_REQUEST).must_next().must_verify(
+            lambda p: {TLV_REQUEST_TLV, NETWORK_DATA_TLV, ACTIVE_TIMESTAMP_TLV} <= set(p.mle.tlv.type))
+
+        # Step 12: Router MUST send a unicast MLE Data Response to SED_1
+        _rpkts.filter_wpan_dst64(SED).filter_mle_cmd(MLE_DATA_RESPONSE).must_next().must_verify(lambda p: {
+            SOURCE_ADDRESS_TLV, LEADER_DATA_TLV, NETWORK_DATA_TLV, ACTIVE_TIMESTAMP_TLV, PENDING_OPERATION_DATASET_TLV
+        } <= set(p.mle.tlv.type) and {
+            NM_CHANNEL_TLV, NM_NETWORK_MESH_LOCAL_PREFIX_TLV, NM_PAN_ID_TLV, NM_DELAY_TIMER_TLV,
+            NM_ACTIVE_TIMESTAMP_TLV
+        } <= set(p.thread_meshcop.tlv.type))
+
+        # Step 14: After NETWORK_ID_TIMEOUT, Router MUST start a new partition
+        _rpkts.filter_ipv6_dst(LINK_LOCAL_ALL_ROUTERS_MULTICAST_ADDRESS).filter_mle_cmd(MLE_PARENT_REQUEST).must_next()
```

Comment:
How to verify it's after`NETWORK_ID_TIMEOUT`  ?

Reply:
isolate Router_1, MED_1, and SED_1 for 300s > 120s, Router_1 will start a new partition

Reply:
I mean beside the test procedure, we should also verify the time gap in the verify?

Reply:
verified the packets time interval > 300 s